### ISSUE TITLE
Can't resolve Spring Boot plugins from plugin repository 

### ISSUE THREAD

**Issue by etiennepeiniau** [gradle/gradle#11380]
### Expected Behavior
When running my build, the Spring Boot plugin declared in the plugins closure should be resolved from the Gradle Plugins Repository.

### Current Behavior
I get an error when my build starts up.

### Context
Without changing anything to our build, we can not retrieve the Spring boot plugin from Gradle Plugin. 

```
.............................................................
Initialized native services in: /root/.gradle/native

Welcome to Gradle 5.4.1!

Here are the highlights of this release:
 - Run builds with JDK12
 - New API for Incremental Tasks
 - Updates to native projects, including Swift 5 support

For more details see https://docs.gradle.org/5.4.1/release-notes.html

To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/5.4.1/userguide/gradle_daemon.html.
Starting process 'Gradle build daemon'. Working directory: /root/.gradle/daemon/5.4.1 Command: /usr/lib/jvm/java-11-openjdk-amd64/bin/java --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.lang.invoke=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError -Xms256m -Xmx512m -Dfile.encoding=UTF-8 -Duser.country -Duser.language=en -Duser.variant -cp /root/.gradle/wrapper/dists/gradle-5.4.1-bin/e75iq110yv9r9wt1a6619x2xm/gradle-5.4.1/lib/gradle-launcher-5.4.1.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 5.4.1
Successfully started process 'Gradle build daemon'
An attempt to start the daemon took 2.209 secs.
The client will now receive all logging from the daemon (pid: 44). The daemon log file: /root/.gradle/daemon/5.4.1/daemon-44.out.log
Daemon will be stopped at the end of the build stopping after processing
Closing daemon's stdin at end of input.
The daemon will no longer process any standard input.
Using 1 worker leases.
Starting Build
Compiling settings file '/root/settings.gradle' using SubsetScriptTransformer.
Compiling settings file '/root/settings.gradle' using BuildScriptTransformer.
Settings evaluated using settings file '/root/settings.gradle'.
Projects loaded. Root project using build file '/root/build.gradle'.
Included projects: [root project 'compute']

> Configure project :
Evaluating root project 'compute' using build file '/root/build.gradle'.
Compiling build file '/root/build.gradle' using SubsetScriptTransformer.
Downloading https://plugins.gradle.org/m2/org/springframework/boot/org.springframework.boot.gradle.plugin/2.1.6.RELEASE/org.springframework.boot.gradle.plugin-2.1.6.RELEASE.pom to /tmp/gradle_download11127669002290407173bin
Resource missing. [HTTP HEAD: https://plugins.gradle.org/m2/org/springframework/boot/org.springframework.boot.gradle.plugin/2.1.6.RELEASE/org.springframework.boot.gradle.plugin-2.1.6.RELEASE.jar]

FAILURE: Build failed with an exception.

* Where:
Build file '/root/build.gradle' line: 5

* What went wrong:
Plugin [id: 'com.adarshr.test-logger', version: '1.6.0'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Plugin Repositories (could not resolve plugin artifact 'com.adarshr.test-logger:com.adarshr.test-logger.gradle.plugin:1.6.0')
  Searched in the following repositories:
    Gradle Central Plugin Repository

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 16s
```

We have tried to build with jdk 11.0.2, 11.0..4 and 11.0.5 with the same result as requested in https://status.gradle.com/incidents/q2vq49fndxg0

This is our plugins block at the top of my build.gradle:

```
plugins {
	id 'org.springframework.boot' version '2.1.6.RELEASE'
	id 'java'
	id 'jacoco'
	id 'com.adarshr.test-logger' version '1.6.0'
	id "org.sonarqube" version "2.7"
}
```

**Comment by ljacomet** [comment]
Not saying that's the issue, but there is something fishy between the plugin blocks you included at the bottom and the version that the build attempts to resolve:
`id 'com.adarshr.test-logger' version '1.7.1'` vs. Plugin [id: 'com.adarshr.test-logger', version: '1.6.0']

And seeing this, I wonder if something else might not be invalid in the setup you used for testing different versions.

Any chance you could do a test with a JDK 11.0.5 with a simpler project?

**Comment by etiennepeiniau** [comment]
Sorry, I did some test on the version of the plugin 'com.adarshr.test-logger' version '1.7.1'.
I've paste the wrong plugins section.

**Comment by ljacomet** [comment]
Thanks, and please run with `--stacktrace` and include the result.

**Comment by etiennepeiniau** [comment]
The problem was solved using the jdk 11.0.5. 
We did not test properly the first time.
Sorry for the disturbance.
